app/exporters/contests/contest_exporter.rb

```diff
@@ -0,0 +1,100 @@
+require 'base_exporter'
+require 'csv'
+
+module Contests
+  class ContestExporter < Exporters::BaseExporter
+    include RenderAnywhere
+
+    def to_csv(contest, include_unranked_contestants)
+      problems = contest.problem_set.problems
+
+      scoreboard = CSV.generate do |csv|
+        csv << ["Rank", "User ID", "Real Name", "Username", "School", "Year"] + problems.map{ |p| p.name } + ["Total Score", "Time"]
+
+        num_ranked = 0
+        rank = nil
+        previous_record = nil
+
+        contest.scoreboard.each do |record|
+          is_ranked = !contest.only_rank_official_contestants || (record.school && record.school_year)
+          next unless is_ranked || include_unranked_contestants
+
+          if is_ranked
+            num_ranked += 1
+            rank = num_ranked unless previous_record && previous_record.score == record.score && previous_record.time_taken == record.time_taken
+            previous_record = record
+          end
+
+          display_rank = is_ranked ? rank : nil
+
+          row = [display_rank, record.user&.id, record.user&.name, record.user&.username, is_ranked ? record.user&.school&.name : nil, is_ranked ? record.school_year : nil]
+          problems.each do |prob|
+            row << record["score_#{prob.id}"]
+          end
+          row += [record.score, format("%d:%02d:%02d",record.time_taken.to_i/3600,record.time_taken.to_i/60%60,record.time_taken.to_i%60)]
+          csv << row
+        end
+      end
+    end
+
+    def export(path, options = {})
+      contest = subject
+      subpath = File.expand_path('submissions', path)
+      dir.mkdir(subpath)
```

Comment:
i may be totally wrong but i thought `dir` had to be `Dir`

Reply:
and a few instances of `file` -> `File` ?

Reply:
It's doing fancy stuff with zip files, see base_exporter.rb.
For context: The code in this PR has already been pushed to the server and works.

Reply:
```      self.dir = Dir
      self.file = File
```

Haha this is pretty unfair on me.

Reply:
okay, had a good read - this is the most "C" like ruby i've ever seen. be worth making it a little more ruby-like down the road, but not an issue for now :)